acorn/src/expression.js

```diff
@@ -856,7 +886,13 @@ pp.initFunction = function(node) {
 
 // Parse object or class method.
 
-pp.parseMethod = function(isGenerator, isAsync, allowDirectSuper) {
+pp.parseFunctionBodyAndFinish = function(node, type, isArrowFunction, isMethod, forInit, inClass) {
```

Comment:
I'd rather have you duplicate a few lines of code in the plugin than us having to support this kind of super-fine-grained split methods forever in the future.